### ISSUE TITLE
CharacterCount not working when multiples editor instances live in the same page

### ISSUE THREAD

**Issue by obetomuniz** [ueberdosis/tiptap#2589]
### What’s the bug you are facing?

`@tiptap/extension-character-count` not working when two or more editors instances live in the same page. 

- The first editor instance does not update unless the second editor instance updates
- When the first editor instance update char count, it updates using the char count of the second editor instance

### How can we reproduce the bug on our side?

1. Create an editor using https://tiptap.dev/api/extensions/character-count extension, remember to display the char count using `editor.storage.characterCount.characters()`
2. Invoke two instances of the previously created editor on the same page
3. Update each editor content

### Can you provide a CodeSandbox?

https://codesandbox.io/s/lively-glade-8h4rc8?file=/src/App.js

### What did you expect to happen?

Each editor instance should update char count separately

### Anything to add? (optional)

Nope

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by obetomuniz** [comment]
Any update on it?

**Comment by cooliopas** [comment]
@obetomuniz did you find a way to fix this? I am having the same issue.

**Comment by obetomuniz** [comment]
@cooliopas Not yet.

**Comment by bdbch** [comment]
@obetomuniz @cooliopas thanks for the issue!

The quickest way to work around is to use `CharacterCount.configure()` instead of `CharacterCount`.

This way stores are completely separated. We'll look into this issue in depth to figure out what caused this weird behaviour.

**Comment by bdbch** [comment]
So the reason why having multiple contexts without using `.configure` is not working right now is because of the way the extensions are created. Right now, when you write your own extension you need to write something like this:

```js
import { Extension } from '@tiptap/core';
const MyAwesomeExtension = Extension.create();
```

The important thing here is, that we are not creating a new instance here but create one global instance of the extension right at the point where the extension is created.

In Extension.ts line 318 we can see, that `Extension.create` is doing the following:
```ts
static create<O = any, S = any>(config: Partial<ExtensionConfig<O, S>> = {}) {
  return new Extension<O, S>(config)
}
```

That means we will always have one object instanced from this Extension class for every editor. The `Extension.configure` function is calling `Extension.extend` which will create a new child extension everytime it's called which is why it's working fine when `.configure` is used.

I referenced this in a new ticket #2694.

@obetomuniz @cooliopas 
I hope for now my temporary solution works fine!

**Comment by obetomuniz** [comment]
Hey @bdbch, it works.

Anyway, it also required to import and pass the `CharacterCount.configure` in the same file that I call `useEditor`:

```javascript
import extensions from "./extensions"
import CharacterCount from "@tiptap/extension-character-count"

const Textarea = () => {
  const editor = useEditor(
    {
      extensions: [...extensions, CharacterCount.configure()],
    },
    []
  )

  return ...
}
```

I mean, all my extensions are grouped in an exportable file, but your approach to making `CharacterCount` apply isolated only works using the way I example above. For now, I will move my extensions to live in the same file, but this weird behavior can also be related, so just reporting it. 

Overall, the workaround suggested works pretty well. Thanks! Let me know if will be better to close this issue or remain with it open until a proper fix for #2694 comes out.
